### ISSUE TITLE
GPU support and true batch processing for SentimentScorer

### ISSUE THREAD

**Issue by caribdigital** [caribdigital/graphhansard#60]
## Problem

`SentimentScorer` in `sentiment.py` has two performance issues:

### 1. Hardcoded CPU-only
Line 132-136: `device=-1` forces CPU inference. On Azure GPU VM with T4, sentiment scoring runs 10-100x slower than necessary.

### 2. Fake batch processing
`score_batch()` (line 176-185) is just a loop over `score()` — no true batching. The transformers pipeline natively supports list input for batch inference.

## Current Code

```python
# CPU only
self.pipeline = pipeline(
    "zero-shot-classification",
    model=self.model_name,
    device=-1,  # CPU only for now
)

# Fake batch
def score_batch(self, contexts: list[str]) -> list[SentimentResult]:
    return [self.score(context) for context in contexts]
```

## Proposed Fix

```python
def __init__(self, model_name: str = "facebook/bart-large-mnli", device: str | None = None):
    self.model_name = model_name
    self._device = device  # None = auto-detect
    self.pipeline = None

def _load_model(self):
    if self.pipeline is None:
        from transformers import pipeline
        import torch
        
        if self._device is None:
            device = 0 if torch.cuda.is_available() else -1
        elif self._device == "cpu":
            device = -1
        else:
            device = 0
        
        self.pipeline = pipeline(
            "zero-shot-classification",
            model=self.model_name,
            device=device,
        )

def score_batch(self, contexts: list[str], batch_size: int = 32) -> list[SentimentResult]:
    self._load_model()
    results = self.pipeline(
        contexts,
        candidate_labels=self._labels,
        multi_label=False,
        batch_size=batch_size,
    )
    return [self._parse_result(r) for r in results]
```

## Files to Modify
- `src/graphhansard/brain/sentiment.py` — `__init__`, `_load_model`, `score_batch`
- `scripts/azure_gpu_process.py` — use `score_batch()` in stage 2 loop

## Impact
- 10-100x speed improvement on GPU
- More consistent results from batch normalization
- Enables processing larger sessions without timeout

## Acceptance Criteria
- [ ] GPU used when available, CPU fallback
- [ ] `device` parameter accepted in constructor
- [ ] `score_batch()` uses pipeline's native list support
- [ ] Backward compatible (default behavior unchanged on CPU)
